### PR TITLE
shipit_code_coverage: Generate static per-test-suite reports weekly

### PR DESCRIPTION
**Pull request by marco-c**

### CODE REVIEW

File: src/shipit_code_coverage/shipit_code_coverage/codecov.py [hunk 1]
```diff
@@ -215,26 +236,45 @@ def go(self):
...
+                run_check(['tar', '-cjf', 'code-coverage-reports/%s.tar.bz2' % suite, suite])
+
+                logger.info('Suite report generated', suite=suite)
+
+            for suite in self.suites:
+                with ThreadPoolExecutorResult() as executor:
+                    executor.submit(lambda: generate_suite_report(suite))
+
+            os.chdir('code-coverage-reports')
+            run_check(['git', 'config', '--global', 'http.postBuffer', '12M'])
```
Comment (La0) [inline src/shipit_code_coverage/shipit_code_coverage/codecov.py]: Why not generate a `.gitconfig` file here ,

Comment (marco-c) [reply src/shipit_code_coverage/shipit_code_coverage/codecov.py]: What's the benefit compared to using `git config`?

**Comment by marco-c** [comment]
@La0 or @jankeromnes, whoever gets to it first, could you review?

File: src/shipit_code_coverage/shipit_code_coverage/codecov.py [hunk 2]
```diff
@@ -145,30 +142,48 @@ def generate_info(self, commit_sha, coveralls_token, suite=None):
...
-          '--commit-sha', commit_sha,
-          '--token', coveralls_token,
         ]
 
-        if suite is not None:
-            cmd.extend(['--service-job-number', str(self.suites.index(suite) + 1)])
-        else:
-            cmd.extend(['--service-job-number', '1'])
+        if out_format == 'coveralls':
+            r = requests.get('https://hg.mozilla.org/mozilla-central/json-rev/%s' % self.revision)
```
Comment (La0) [inline src/shipit_code_coverage/shipit_code_coverage/codecov.py]: Maybe check the status code
```
r.raise_for_status()
```
or usually i do
```
assert r.ok, "Failed to retrieve mozilla revision {}".format(self.revision)
```

File: src/shipit_code_coverage/shipit_code_coverage/codecov.py [hunk 3]
```diff
@@ -258,3 +241,60 @@ def go(self):
...
+
+            os.chdir('code-coverage-reports')
+            run_check(['git', 'config', '--global', 'http.postBuffer', '12M'])
+            run_check(['git', 'config', '--global', 'user.email', 'report@upload.it'])
+            run_check(['git', 'config', '--global', 'user.name', 'Report Uploader'])
+            repo_url = 'https://%s:%s@github.com/marco-c/code-coverage-reports' % (self.gecko_dev_user, self.gecko_dev_pwd)
+            run_check(['git', 'init'])
+            run_check(['git', 'add', '*'])
+            run_check(['git', 'commit', '-m', 'Coverage reports upload'])
+            retry(lambda: run_check(['git', 'push', repo_url, 'master', '--force']))
```
Comment (La0) [inline src/shipit_code_coverage/shipit_code_coverage/codecov.py]: :+1:  this syntax.

File: src/shipit_code_coverage/default.nix [hunk 4]
```diff
@@ -63,6 +63,7 @@ let
       hook = mkTaskclusterHook {
         name = "Shipit task aggregating code coverage data";
         owner = "mcastelluccio@mozilla.com";
+        schedule = [ "0 0 0 * * 0" ]; # every week
```
Comment (La0) [inline src/shipit_code_coverage/default.nix]: I checked this one as the nodejs cron format is tricky, this is indeed once per week. :+1:

File: src/shipit_code_coverage/shipit_code_coverage/codecov.py (hunk 2 above)
Comment (marco-c) [reply src/shipit_code_coverage/shipit_code_coverage/codecov.py]: Good idea, I'm calling `raise_for_status` now.

**Comment by marco-c** [comment]
> Build+Checks are fine on my computer.
Just an optional nit. Good work

Thanks, I've fixed the nit!

**Pull request merged**